### PR TITLE
add test workflow

### PR DESCRIPTION
**Pull request by Saza-ku**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+name: Test
```
Comment (shintard) [inline .github/workflows/test.yml:1]: [nits] i guess may be split into multiple tests in the future? I hope we can make it easy to understand and name them then.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+on:
+  push:
+
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-22.04
```
Comment (shintard) [inline .github/workflows/test.yml:9]: I don't think it needs to be 22.04, so I think `ubuntu-latest` is fine, hard coding the version makes me forget to manage it, so basically I'd rather use `latest`

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      
+      - name: Setup GitHub token
+        run: |
+          git config --global url."https://${{ secrets.GH_PAT_REPO }}:x-oauth-basic@github.com/".insteadOf "https://github.com/"
+
+      - name: Get dependencies
+        run: |
+            go mod download -x
+
+      - name: Test
```
Comment (shintard) [inline .github/workflows/test.yml:33]: `Go Test` ?

**Review by shintard — COMMENTED** [review]
@Saza-ku checked!

File: .github/workflows/test.yml (hunk 1 above)
Comment (Saza-ku) [reply .github/workflows/test.yml:1]: > i guess may be split into multiple tests in the future?

Like unit tests, integration tests, and e2e tests?

So this name should be `Unit Test`, right?

Comment (shintard) [reply .github/workflows/test.yml:1]: right, you can change the `UnitTest`

**Comment by shintard** [comment]
@Saza-ku reviewed!

**Comment by shintard** [comment]
lgtm!

**Pull request merged**